src/event.c

```diff
@@ -506,7 +506,7 @@ static inline void ev_property_notify(session_t *ps, xcb_property_notify_event_t
 	}
 
 	// If frame extents property changes
-	if (ps->o.frame_opacity > 0 && ev->atom == ps->atoms->a_NET_FRAME_EXTENTS) {
+	if (ev->atom == ps->atoms->a_NET_FRAME_EXTENTS) {
```

Comment:
Is _NET_FRAME_EXTENTS used for something other than frame_opacity?

Reply:
It is also used to determine wether the window has a frame with potential transparency / alpha even when `frame-opacity` is not set (i.e. defaults to 1) when checking the internal window mode. See https://github.com/yshui/picom/commit/97db599fd8726235beae62391724ad26aa0a0856 that changed to handle changes even for `frame-opacity = 1`.

Reply:
Ah, thanks!